### PR TITLE
Move log tokenizer to `lib/strvals` package

### PR DESCRIPTION
**Pull request by ka3de**
## What?

Moves the tokenizer implementation, currently only used to parse the log output option configuration, to a more generic package.

## Why?

Up until now the tokenizer implementation was only used to parse the log output option configuration. With our current intention of using this format for other options, such as traces output, it becomes evident that this implementation should be moved away from the log package and into a more generic one.

## Checklist

- [X] I have performed a self-review of my code.
- [ ] I have added tests for my changes. **(N/A)**
- [X] I have run linter locally (`make lint`) and all checks pass.
- [X] I have run tests locally (`make tests`) and all tests pass.
- [X] I have commented on my code, particularly in hard-to-understand areas.

## Related PR(s)/Issue(s)

Related #3445

### CODE REVIEW

**Review by mstoykov — APPROVED** [review]
LGTM!

My only nit is that I kind of don't like `parseutils` in part due to the `utils` part.

But `tokenizer.Tokenize` is no better as well as `tokens.Parse` or `tokens.New`

So unless @oleiade comes up wiht something better I guess this is it.

**Comment by ka3de** [comment]
> My only nit is that I kind of don't like parseutils in part due to the utils part

Yep, to be honest I don't quite like it either. I considered `parse` or `parseutil`, but because there was already a `testutils` package, decided to be coherent with that. But open to any suggestion for another name.

**Comment by oleiade** [comment]
I like `strvals` 👍🏻 Let's go for that if you're okay with it @ka3de 🙇🏻

**Comment by ka3de** [comment]
> My reasoning is that the parsing mechanism described in the package is quite specific to the key=value,... format, which is meant to be used in logs, and I don't think that calling parseutils.Tokenize is very clear on what to expect from it

I guess this depends also on whether this package will hold more parsing implementations in the future. But as of now I agree that the implementation is pretty specific.

> I would vote for making the name of the package as specific and explicit as possible, rather than broad and generic. Thus, I would suggest something along the lines of lib/logparser, or lib/kvparser package or something along those lines instead.

My understanding is that the format implemented by `Tokenize` is not strictly related with log/s. We are actually now, with this PR, using it to parse the _traces output_ configuration. So to me `kvparser` is a better fit as of now :+1:

File: lib/strvals/parser.go [hunk 1]
```diff
@@ -1,9 +1,10 @@
-package log
+package strvals
 
 import "fmt"
 
-type token struct {
-	key, value string
+// Token represents a key-value token.
+type Token struct {
```
Comment (ka3de) [inline lib/strvals/parser.go:6]: After the renaming to `strvals` package and `Parse` method, I left this as `Token` as the definition has not changed, but I could also see this and `tokenizer` changing to:
```
s/Token/KV
s/tokenizer/parser
```

**Review by mstoykov — APPROVED** [review]
LGMT!

Just rename the PR title before merge ;)

File: lib/strvals/parser.go (hunk 1 above)
Comment (mstoykov) [reply lib/strvals/parser.go:6]: The latter can definitely be changed later as it is not exported.

I am don't think `Token` is not specific enough - so it is okay IMO.

**Review by oleiade — APPROVED** [review]

**Comment by ka3de** [comment]
I have no permission to merge the PR. Could you merge this please?

**Pull request merged**